linera-views/src/log_view.rs

```diff
@@ -190,6 +190,46 @@ where
         Ok(value)
     }
 
+    /// Reads several logged keys (including staged ones)
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::memory::create_test_context;
+    /// # use linera_views::log_view::LogView;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_test_context();
+    ///   let mut log = LogView::load(context).await.unwrap();
+    ///   log.push(34);
+    ///   log.push(42);
+    ///   assert_eq!(log.multi_get(vec![0,1]).await.unwrap(), vec![Some(34),Some(42)]);
+    /// # })
+    /// ```
+    pub async fn multi_get(&self, indices: Vec<usize>) -> Result<Vec<Option<T>>, ViewError> {
+        let mut result = Vec::new();
+        if self.was_cleared {
+            for index in indices {
+                result.push(self.new_values.get(index).cloned());
+            }
+        } else {
+            let mut keys = Vec::new();
+            let mut positions = Vec::new();
+            for (pos, index) in indices.into_iter().enumerate() {
+                if index < self.stored_count {
+                    let key = self.context.derive_tag_key(KeyTag::Index as u8, &index)?;
+                    keys.push(key);
+                    positions.push(pos);
+                    result.push(None);
+                } else {
+                    result.push(self.new_values.get(index - self.stored_count).cloned());
+                }
+            }
+            let values = self.context.read_multi_key(keys).await?;
+            for (pos, value) in positions.into_iter().zip(values) {
+                *result.get_mut(pos).unwrap() = value;
```

Comment:
Doesn't this `unwrap()` always panic? Because `positions` contains exactly those where we pushed `None`? Did you mean:
```suggestion
                *result.get_mut(pos) = Some(value);
```

Reply:
No.
The `get_mut` returns an `option<&mut val>` because the index could well be missing. The `unwrap()` does not fail because we have built the vector before.
On the other hand, the `read_key` returns an `option<V>` because the value can be present or not in the database (similarly for read_multi_key).

Reply:
Right, thanks, I got confused there!